internal-services/catalog/file-updates-process-file-updates-task.yaml

```diff
@@ -72,14 +72,21 @@ spec:
             cd $(workspaces.temp.path)
             git_clone_and_checkout --repository $REPO --revision $REVISION
 
-            update_tmp=$(mktemp).yaml
             for file in ${FILES}; do
-                jq -cr ".paths[] | select(.path==\"${file}\") | .replacements[]" $TEMP| \
+
+                # get the extra newlines that `yq` will ignore
+                OVERHEAD=`grep ^$ $file | wc -l`
+
+                jq -cr ".paths[] | select(.path==\"${file}\") | .replacements[] | [.key, .replacement] | join(\"\t\")" $TEMP| \
                 while read replacement; do
-                   REPLACEMENT=$(jq -cr '[.yamlpath, .replace] | join(" ")' <<< $replacement)
-                   read YAMLPATH REPLACE <<< $REPLACEMENT
-                   LN=`python3 /home/utils/yamlline.py --file $file --path $YAMLPATH`
-                   LINES=$(echo -e `yaml-get -p "$YAMLPATH" $file` | wc -l)
+                   IFS=$'\t' read KEY REPLACE <<< $replacement
+                   # the key position
+                   LN=`yq "${KEY} | line" $file`
+
+                   # the number of lines + overhead to run the replace 
+                   LINES=$(expr `yq "${KEY}" $file |wc -l` + ${OVERHEAD})
```

Comment:
Am I reading this right that you don't care if the empty lines are before our field or after? You will just add as many extra lines to the range as the number of empty lines you find, just to be sure to cover the section you want? But that also means that you may include some lines you don't want, right? Which could be an issue - the sed could change something other than what you want.

Reply:
Indeed. I will change to only add the blank newlines until the position. I thought about that yesterday but forgot to make the changes. Thanks for saying it!

Reply:
Noticed now that actually `yq` is giving the position but with just 1 line less for all cases. I tested  several different keys  in the bottom of the that deploy file and still just 1 line of difference. Please take a look.

Reply:
I don't think that's the case. For example I have a file like this:
```yaml







resourceTemplates:
 - name: osd-fleet-manager-production
 ```

And I get:

```sh
$ cat test.yaml | yq '.resourceTemplates[0] | line'
2
```

Reply:
We cleared this up.